Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -8,28 +8,29 @@
 
 namespace UnityEngine.InputSystem.Editor
 {
+    class ProjectWideActionsAssetPostprocessor : AssetPostprocessor
+    {
+        static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths, bool didDomainReload)
+        {
+            if (ArrayHelpers.Contains(importedAssets, ProjectWideActionsAsset.kAssetPath))
+            {
+                AssetDatabase.DisallowAutoRefresh();
+                ProjectWideActionsAsset.CreateRoslynAdditionalFile();
+                AssetDatabase.AllowAutoRefresh();
+            }
+        }
+    }
+
     internal static class ProjectWideActionsAsset
     {
-        internal const string kDefaultAssetPath = "Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsTemplate.json";
+        internal const string kTemplateAssetPath = "Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsTemplate.json";
         internal const string kAssetPath = "ProjectSettings/InputManager.asset";
         internal const string kAssetName = InputSystem.kProjectWideActionsAssetName;
 
-        static string s_DefaultAssetPath = kDefaultAssetPath;
-        static string s_AssetPath = kAssetPath;
+        internal const string kAdditionalFilePath = "Assets/actions.InputSystemActionsAPIGenerator.additionalfile"; // Copy of asset that is fed to the SourceGenerator
 
 #if UNITY_INCLUDE_TESTS
-        internal static void SetAssetPaths(string defaultAssetPath, string assetPath)
-        {
-            s_DefaultAssetPath = defaultAssetPath;
-            s_AssetPath = assetPath;
-        }
-
-        internal static void Reset()
-        {
-            s_DefaultAssetPath = kDefaultAssetPath;
-            s_AssetPath = kAssetPath;
-        }
-
+        internal static InputActionAsset testAsset  { get; set; }
```

Comment:
I am afraid just returning another InputActionAsset here might just be a partial workaround for tests depending on what they are covering. There is no code to replace / swap out object in AssetDatabase added via AddObjectToAsset (references and InputActionAsset) so any attempt to access via non-instance or doing edits would cause a mixed perspective of reality. Maybe this is ok for now, but just want to highlight that if we really want to properly substitute the InputActionAsset we might have do more work. Possibly I guess we could remove this property and modify/restore the AssetDatabase from Test fixture instead which would eliminate this property and if statement below. Maybe its as simple as deleting the asset object from the path via ADB (but that could remove local changes) or extract instance, modify ADB and restore?

In test TearDown:
Restore InputActionAsset by invoking AssetDatabase.AddObjectToPath(kAssetPath);